crowdfunding/projects/views.py

```diff
@@ -1,3 +1,90 @@
 from django.shortcuts import render
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from django.http import Http404
+from rest_framework import status, generics, permissions
 
-# Create your views here.
+from .models import Project, Pledge
+from .serializers import ProjectSerializer, PledgeSerializer, ProjectDetailSerializer
+from .permissions import IsOwnerOrReadOnly
+# from datetime import datetime, timedelta
+
+class ProjectList(APIView):
+    permission_classes = [permissions.IsAuthenticatedOrReadOnly, IsOwnerOrReadOnly]
```

Comment:
I'm not sure you need is ower of readonly on the get/create view, as this  won't really apply at create time.